### DISCUSSION TITLE
DIRECTUS - on creating a new Collection Primary Key (id) does not get added

### DISCUSSION THREAD

**Discussion by rflieb** [directus/directus#18047] [Archive]
Dear Community
in Directus 9.23.1 - we have the error/bug that on creating a new collection the primary key (id) is not created as it should be / was.
This happened out of the blue.

If we want to access the collection then (without primary key (id)) - we get an error >> Permission Denied

![Screenshot 2023-04-04 at 10 01 49](https://user-images.githubusercontent.com/78856756/229727738-c9a19acf-00a0-42ef-b98f-bebc9c76a302.png)

![Screenshot 2023-04-04 at 10 01 58](https://user-images.githubusercontent.com/78856756/229727790-89bb71a7-5bfc-45ac-a0fe-d413e0ae8b52.png)

**UPDATE:**

it seems it is not the primary key - it is the role handling.
Although I am admin, I have no permissions to create new created collections.

Any thoughts on this?
Thank you

Ralf

**Comment by JustRoR** [comment]
I've run into the same problem. I think this is the same primary key issue Asheroto experienced here: [#15233](https://github.com/directus/directus/discussions/15233)

In my case, we're running Directus along Shopware 6.5, and this error occurs on tables where there are two primary keys (one a composite primary key). 

I haven't found a solution yet unfortunately.
